test/freeparameterexpression.jl

```diff
@@ -0,0 +1,70 @@
+using Braket, Test
+
+@testset "Free parameter expressions" begin
+    α = FreeParameter(:alpha)
+    θ = FreeParameter(:theta)
+    gate = FreeParameterExpression("α + 2*θ")
+    @test copy(gate) === gate
+    gsub = subs(gate, Dict(:α => 2.0, :θ => 2.0))
+    circ = Circuit()
+    circ = H(circ, 0)
+    circ = Rx(circ, 1, gsub)
```

Comment:
But here `gsub` is still the FPE with the values subbed in. What happens if I do:

```
circ = Rx(circ, 1, gate)
```

?

Reply:
That will show the  FPE before evaluating it via substitute at the backend wrapped as subs:

```
julia> circ = Circuit()

T : |Result Types|
                
T : |Result Types|

julia> circ = H(circ, 0)
T  : |0|Result Types|
                     
q0 : -H--------------
                     
T  : |0|Result Types|


julia> circ = Rx(circ, 1, gate)
T  : |    0     |Result Types|
                              
q0 : -H-----------------------
                              
q1 : -Rx(α + 2θ)--------------
                              
T  : |    0     |Result Types|


```